packages/fast-components-react-msft/src/action-toggle/action-toggle.tsx

```diff
@@ -0,0 +1,239 @@
+import * as React from "react";
+import { get } from "lodash-es";
+import Foundation, { HandledProps } from "@microsoft/fast-components-foundation-react";
+import { Button, ButtonAppearance } from "../button";
+import {
+    ActionToggleAppearance,
+    ActionToggleHandledProps,
+    ActionToggleProps,
+    ActionToggleUnhandledProps,
+} from "./action-toggle.props";
+import { actionToggleButtonOverrides } from "@microsoft/fast-components-styles-msft";
+import { isNullOrWhiteSpace } from "../../../fast-web-utilities/dist";
+import { isNullOrUndefined } from "util";
+
+export interface ActionToggleState {
+    selected: boolean;
+}
+
+class ActionToggle extends Foundation<
+    ActionToggleHandledProps,
+    ActionToggleUnhandledProps,
+    ActionToggleState
+> {
+    public static displayName: string = "ActionToggle";
+
+    /**
+     * React life-cycle method
+     */
+    public static getDerivedStateFromProps(
+        nextProps: ActionToggleProps,
+        prevState: ActionToggleState
+    ): null | Partial<ActionToggleState> {
+        if (
+            typeof nextProps.selected === "boolean" &&
+            nextProps.selected !== prevState.selected
+        ) {
+            return {
+                selected: nextProps.selected,
+            };
+        }
+
+        return null;
+    }
+
+    protected handledProps: HandledProps<ActionToggleHandledProps> = {
+        appearance: void 0,
+        managedClasses: void 0,
+        disabled: void 0,
+        selected: void 0,
+        selectedGlyph: void 0,
+        unselectedGlyph: void 0,
+        selectedText: void 0,
+        unselectedText: void 0,
+        selectedARIALabel: void 0,
+        unselectedARIALabel: void 0,
+    };
+
+    /**
+     * Define constructor
+     */
+    constructor(props: ActionToggleProps) {
+        super(props);
+
+        this.state = {
+            selected: this.props.selected || false,
+        };
+    }
+
+    /**
+     * Renders the component
+     */
+    public render(): JSX.Element {
+        return (
+            <Button
+                {...this.unhandledProps()}
+                className={this.generateClassNames()}
+                disabled={this.props.disabled}
+                onClick={this.handleToggleChange}
+                aria-label={this.renderARIALabel()}
+                appearance={
+                    ButtonAppearance[ActionToggleAppearance[this.props.appearance]]
+                }
+                jssStyleSheet={actionToggleButtonOverrides}
+            >
+                {this.renderGlyph()}
+                {this.renderLabel()}
+                {this.props.children}
+            </Button>
+        );
+    }
+
+    /**
+     * Returns the appropriate ARIA label
+     */
+    public renderARIALabel(): string {
```

Comment:
Curious as to @nicholasrice and @janechu's thoughts on this, but I wonder if it doesn't make more sense to simply manage this as a completely controlled component. On change we may would expect new values for glyph and children. Does it makes sense for this to support controlled behavior only?

Reply:
I think I'm on board with it being controlled.

However, I don't think that means we shouldn't have selected/unselected props (I think that is what you're proposing above with only accepting children). Just taking children, I now need to manage when what child text is applied which at some level defeats the point of the component. Take that further to glyphs and aria-labels, and the component is pretty much useless.

I think the only "controlled" prop should be `selected` - the other info is provided and the component determines what to display based on that state.

Reply:
Sounds fair. 

If we are limiting this to selected/unselected text (strings) I think we convolute by passing children as well. If this is a pattern/subset/more restricted design system component - does it need "children" AND selected/unselected text? If I want to do something special outside of this pattern I can use the button component to do so. 

If we want to support passing components here as well and that is why we still support children, I think we would be better served somehow making the props selectedContent and unselectedContent and accepting a generic React node. Thoughts?

Reply:
No longer support children.  Content is a react node rather than just a string.